File: app/src/main/java/au/id/blackwell/kurt/lantv/player/TvMediaPlayer.java
package au.id.blackwell.kurt.lantv.player;

import android.view.SurfaceHolder;

import java.io.IOException;

import au.id.blackwell.kurt.lantv.MediaDetails;

interface TvMediaPlayer {
    boolean isPlaying();
    void setListener(TvMediaPlayerListener listener);
    void setMedia(MediaDetails details) throws IOException;
    void setScreenOnWhilePlaying(boolean onWhilePlaying);
    void setDisplay(SurfaceHolder holder);
    void prepareAsync();
    void release();
    void start();
    void pause();
}

File: app/src/main/java/au/id/blackwell/kurt/lantv/player/TvMediaPlayer.kt
package au.id.blackwell.kurt.lantv.player

import android.view.SurfaceHolder

import java.io.IOException

interface TvMediaPlayer {
    fun isPlaying(): Boolean
    fun setListener(listener: TvMediaPlayerListener)
    @Throws(IOException::class)
    fun setMedia(details: MediaDetails)

    fun setScreenOnWhilePlaying(onWhilePlaying: Boolean)
    fun setDisplay(holder: SurfaceHolder?)
    fun prepareAsync()
    fun release()
    fun start()
    fun pause()
}
